WPMedia/engine-theme-sdk
# PEN-1298: Add Beta release Github action (Step 2)

## Description
Added a Github action to automate Engine Theme SDK beta releases. This runs on every push to the beta branch.
Relevant documentation will be updated on later steps of this ticket

## Jira Ticket
- [PEN-1298](https://arcpublishing.atlassian.net/browse/PEN-1298)

## Acceptance Criteria
On merge into stable branch name, should publish to stable tag. Same with canary and beta branches and tags 

## Test Steps
Push or make a PR against the beta branch and see the action being triggered and successfully publishing a new Engine Theme SDK version (run `npm view @wpmedia/engine-theme-sdk` to verify)

## Effect Of Changes
### Before
Engine Theme SDK beta releases have to be done by manually running publishing commands 

### After
Engine Theme SDK beta releases are now done automatically through a Github action. This reduces chances of errors like a forgotten local tag that wasn't pushed.

## Dependencies or Side Effects
None

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
